internal/core/src/segcore/Reduce.cpp

```diff
@@ -417,4 +479,71 @@ ReduceHelper::GetSearchResultDataSlice(int slice_index) {
     return buffer;
 }
 
+void
+ReduceHelper::AssembleGroupByValues(
+    std::unique_ptr<milvus::proto::schema::SearchResultData>& search_result,
+    const std::vector<GroupByValueType>& group_by_vals) {
+    auto group_by_field_id = plan_->plan_node_->search_info_.group_by_field_id_;
+    if (group_by_field_id.has_value() && group_by_vals.size() > 0) {
+        auto group_by_values_field =
+            std::make_unique<milvus::proto::schema::ScalarField>();
+        auto group_by_field =
+            plan_->schema_.operator[](group_by_field_id.value());
+        DataType group_by_data_type = group_by_field.get_data_type();
+        for (std::size_t idx = 0; idx < group_by_vals.size(); idx++) {
```

Comment:
swap the order of `for` and `switch`, then a batch append would be possible, which could improve the performance